File: sample-java/src/main/java/com/smart/github/app/di/ActivityModule.java
package com.smart.github.app.di;

import com.smart.github.app.trend.TrendActivity;
import dagger.Module;
import dagger.android.AndroidInjectionModule;
import dagger.android.ContributesAndroidInjector;

@Module(includes = AndroidInjectionModule.class)
public abstract class ActivityModule {

    @ContributesAndroidInjector
    abstract TrendActivity bindMainActivity();
}

File: sample/src/main/java/com/smart/sample/app/di/ActivityModule.kt
package com.smart.sample.app.di

import com.smart.sample.app.trend.TrendActivity
import dagger.Module
import dagger.android.AndroidInjectionModule
import dagger.android.ContributesAndroidInjector

@Module(includes = [AndroidInjectionModule::class])
abstract class ActivityModule {

    @ContributesAndroidInjector
    abstract fun bindMainActivity(): TrendActivity
}